Libraries/Opc.Ua.Server/Diagnostics/CustomNodeManager.cs

```diff
@@ -3457,55 +3416,53 @@ public virtual void CreateMonitoredItems(
             List<NodeHandle> nodesToValidate = new List<NodeHandle>();
             List<IMonitoredItem> createdItems = new List<IMonitoredItem>();
 
-            lock (Lock)
+
+            for (int ii = 0; ii < itemsToCreate.Count; ii++)
             {
-                for (int ii = 0; ii < itemsToCreate.Count; ii++)
-                {
-                    MonitoredItemCreateRequest itemToCreate = itemsToCreate[ii];
+                MonitoredItemCreateRequest itemToCreate = itemsToCreate[ii];
 
-                    // skip items that have already been processed.
-                    if (itemToCreate.Processed)
-                    {
-                        continue;
-                    }
+                // skip items that have already been processed.
+                if (itemToCreate.Processed)
+                {
+                    continue;
+                }
 
-                    ReadValueId itemToMonitor = itemToCreate.ItemToMonitor;
+                ReadValueId itemToMonitor = itemToCreate.ItemToMonitor;
 
-                    // check for valid handle.
-                    NodeHandle handle = GetManagerHandle(systemContext, itemToMonitor.NodeId, operationCache);
+                // check for valid handle.
+                NodeHandle handle = GetManagerHandle(systemContext, itemToMonitor.NodeId, operationCache);
 
-                    if (handle == null)
-                    {
-                        continue;
-                    }
+                if (handle == null)
+                {
+                    continue;
+                }
 
-                    // owned by this node manager.
-                    itemToCreate.Processed = true;
+                // owned by this node manager.
+                itemToCreate.Processed = true;
 
-                    // must validate node in a separate operation.
-                    errors[ii] = StatusCodes.BadNodeIdUnknown;
+                // must validate node in a separate operation.
+                errors[ii] = StatusCodes.BadNodeIdUnknown;
 
-                    handle.Index = ii;
-                    nodesToValidate.Add(handle);
-                }
+                handle.Index = ii;
+                nodesToValidate.Add(handle);
+            }
 
-                // check for nothing to do.
-                if (nodesToValidate.Count == 0)
-                {
-                    return;
-                }
+            // check for nothing to do.
+            if (nodesToValidate.Count == 0)
+            {
+                return;
             }
 
-            // validates the nodes (reads values from the underlying data source if required).
-            for (int ii = 0; ii < nodesToValidate.Count; ii++)
+            lock (Lock)
             {
```

Comment:
this lock was taken with higher granularity on the inside, did you move it on purpose?